### DISCUSSION TITLE
v2/zh/cookbook/advanced/extend

### DISCUSSION THREAD

**Discussion by giscus** [vuepress-theme-hope/giscus-discussions#394] [Announcements]
# v2/zh/cookbook/advanced/extend

vuepress-theme-hope 和默认主题一样，同样支持继承。 你可以根据自己的需求在 vuepress-theme-hope 的基础上二次创作自己的主题并本地使用或发布它。

https://vuepress-theme-hope.github.io/v2/zh/cookbook/advanced/extend.html

**Comment by Mrceel** [comment]
extends: hopeTheme(options),
这里的options是什么？ 个人配置？

**Reply by Mister-Hope** [reply]
主题原来的配置，你应该把它pass through进去。

**Comment by qq584686739** [comment]
对于主题的扩展，应该是很多人都会做的，比如，在现有的页面上添加一些自己喜欢的模块，或者是添加一些背景元素等等。

我认为这里说的过于简单，导致许多人应该扩展失败。我举个简单例子：官方可以自定义主题，然后使用父主题的Layout.vue进行引入然后使用插槽进行扩展，在这里我应该是没有找到任何插槽的东西，简单的去翻了一下源码，比较难看懂，相反官方的默认主题vue形式很通俗易懂。

我建议作者能够完善这个扩展模块。

**Reply by qq584686739** [reply]
举个例子，官方默认主题上，我想新加一个div模块，我可以使用插槽，修饰样式后就可以了。

如果是hope主题，我现在看完文档，我不知道如何做才能达到不影响现有的东西进行扩展。

**Reply by Mister-Hope** [reply]
![image](https://user-images.githubusercontent.com/33315834/207580236-a1d57dd9-4f19-4a69-8009-c599b27240d5.png)

所以请问你看这页了么？
